src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php

```diff
@@ -133,7 +133,10 @@ public function publish(string $body, array $headers = []): void
             $this->setup();
         }
 
-        $this->exchange()->publish($body, $this->queueConfiguration['routing_key'] ?? null, AMQP_NOPARAM, ['headers' => $headers]);
+        $flags = $this->queueConfiguration['flags'] ?? AMQP_NOPARAM;
+        $attributes = array_merge_recursive($this->queueConfiguration['attributes'] ?? [], ['headers' => $headers]);
```

Comment:
I'm not sure it needs to be recursive.

Reply:
If we want to send some default headers through attributes, then I think it needs. In general we can use array_merge

Reply:
Can you add a test case where this is needed please?

Reply:
> Can you add a test case where this is needed please?

updated test case